central/hash/manager/deduper.go

```diff
@@ -71,6 +72,19 @@ type deduperImpl struct {
 	hasher *hash.Hasher
 }
 
+func pruneInvalidAlerts(existingHashes map[string]uint64) {
+	// Due to a bug where we are not skipping deduping on ATTEMPTED alerts
```

Comment:
@connorgorman can you elaborate on this?

Reply:
In the hash deduper, we were treating attempted alerts as normal alerts that could be potentially deduped even though they can't. This would lead to the "hash" maps growing until we hit max number of hashes and reseting. This ensures that we remove them from the "hash" maps because the only valid alerts are ones in which the id of the alert result is the same id as a deployment

Reply:
This does not cause an issue with the processing of the attempted alerts as they will always pass the deduper, but does potentially cause problems in the stored hashes

Reply:
It would be great if your description comment in this thread is added as the code comment.